controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook_test.go

```diff
@@ -293,6 +291,32 @@ func TestKubeadmControlPlaneValidateUpdate(t *testing.T) {
 	scaleToEvenExternalEtcdCluster := beforeExternalEtcdCluster.DeepCopy()
 	scaleToEvenExternalEtcdCluster.Spec.Replicas = pointer.Int32Ptr(2)
 
+	beforeInvalidEtcdCluster := before.DeepCopy()
+	beforeInvalidEtcdCluster.Spec.KubeadmConfigSpec.ClusterConfiguration = &kubeadmv1beta1.ClusterConfiguration{
+		Etcd: kubeadmv1beta1.Etcd{
+			Local: &kubeadmv1beta1.LocalEtcd{
+				ImageMeta: kubeadmv1beta1.ImageMeta{
+					ImageRepository: "image-repository",
+					ImageTag:        "latest",
+				},
+			},
+		},
+	}
+	afterInvalidEtcdCluster := beforeInvalidEtcdCluster.DeepCopy()
+	afterInvalidEtcdCluster.Spec.KubeadmConfigSpec.ClusterConfiguration = &kubeadmv1beta1.ClusterConfiguration{
+		Etcd: kubeadmv1beta1.Etcd{
+			External: &kubeadmv1beta1.ExternalEtcd{
+				Endpoints: []string{"127.0.0.1"},
+			},
+			Local: &kubeadmv1beta1.LocalEtcd{
```

Comment:
this local should be undefined in this struct. The incoming would only have the external defined and our validations should catch that there is already local defined and we're trying to also define an external configuration.